source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -81,87 +84,81 @@ Upgrading Mattermost Server
 
-10. Clear the contents of this directory. If you've added custom directories to the command, be sure to add those to the following command. For example:
+  .. code-block:: sh
 
-.. code-block:: sh
+    sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data \) -prune \) | sort | sudo xargs rm -r  
+  
+  **Using Bleve Search**
 
-  sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data \) -prune \) | sort | sudo xargs rm -r
+  If using `Bleve Search <https://docs.mattermost.com/deploy/bleve-search.html>`__, and the directory exists *within* the ``mattermost`` directory, the index directory path won't be preserved with the commands below. 
+  
+  - You can either move the bleve index directory out from the ``mattermost`` directory before upgrading or, following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory. 
+  - You can then store that directory or re-index as preferred. 
+  - The bleve indexes can be migrated without reindexing between Mattermost versions. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
       
-11. Change ownership of the new files before copying them. For example:
+8. Change ownership of the new files before copying them. For example:
 
-.. code-block:: sh
+  .. code-block:: sh
          
-  sudo chown -hR mattermost:mattermost /tmp/mattermost-upgrade/
+    sudo chown -hR mattermost:mattermost /tmp/mattermost-upgrade/
      
 .. note::
     
   - If you didn't use ``mattermost`` as the owner and group of the install directory, run ``sudo chown -hR {owner}:{group} tmp/mattermost-upgrade/``.
   - If you're uncertain what owner or group was defined, use the ``ls -l {install-path}/mattermost/bin/mattermost`` command to obtain them.
 
-12. Copy the new files to your install directory and remove the temporary files.
+9. Copy the new files to your install directory and remove the temporary files.
 
-.. code-block:: sh
+  .. code-block:: sh
 
-  sudo cp -an /tmp/mattermost-upgrade/. mattermost/
-  sudo rm -r /tmp/mattermost-upgrade/
-  sudo rm -i /tmp/mattermost*.gz
+   sudo cp -an /tmp/mattermost-upgrade/. mattermost/
+   sudo rm -r /tmp/mattermost-upgrade/
+   sudo rm -i /tmp/mattermost*.gz
 
-.. note::
-  The ``n`` (no-clobber) flag and trailing ``.`` on source are very important.
+  .. note::
+    The ``n`` (no-clobber) flag and trailing ``.`` on source are very important.
 
```

Comment:
*The ``n`` (no-clobber) flag preserves existing configs and logs in your installation path and the trailing ``.`` on source ensures all install files are copied.*? - This makes it more clear and less opaque.